solr/core/src/java/org/apache/solr/search/SolrIndexSearcher.java

```diff
@@ -1299,6 +1302,41 @@ public DocList getDocList(Query query, List<Query> filterList, Sort lsort, int o
   public static final int GET_DOCLIST = 0x02; // get the documents actually returned in a response
   public static final int GET_SCORES = 0x01;
 
+  private static boolean isConstantScoreQuery(Query q) {
+    if (q instanceof BoostQuery) {
+      // ConstantScoreQueries are often (always?) wrapped in BoostQuery to assign specific score
```

Comment:
I suspect this partially goes away after #529

Reply:
Something to keep an eye out for; I applied a patch for #529 and this wrapping still seems to happen. I think this is a side-effect of the way "ConstantScoreQuery" gets specified in queries (`(clause)^=n`) where `n` is the constant score, which is applied via wrapping in a BoostQuery. In any event it doesn't look like that changes with #529

Reply:
Ok, thanks for checking. We might also want to check for `WrappedQuery`? Not sure.